### ISSUE TITLE
[Feature request] integration with DeepConf

### ISSUE THREAD

**Issue by itsmeknt** [codelion/optillm#230]
You've probably seen it already, but the [DeepConf](https://arxiv.org/abs/2508.15260) paper was released earlier this week. It seems to be the only inference-time scaling solution that achieved 99.9% AIME score with open weight models (GPT-OSS 120B).

For reference, [here](https://github.com/pnocera/llama.cpp/blob/master/docs/deepconf.md) is a llama.cpp fork that supports DeepConf

Do you have any plans to add this to optillm anytime soon?

If not, I'm happy to give a go at implementing this and submitting a PR. Will be my first time hacking optillm, so this might take ~2-3 weeks

**Comment by codelion** [comment]
I added an initial implementation here - https://github.com/codelion/optillm/tree/main/optillm/deepconf you can help test and evaluate it.

**Comment by itsmeknt** [comment]
Happy to test it! Would it be helpful if I run it against some benchmarks and share the results too? I can only run small models that can fit in a single 3090, but I'm curious about the improvement

**Comment by codelion** [comment]
Sure, you can test with existing benchmarks that are already there in the repo AIME-24, MATH-500, OptiLLMBench and ArenaHardAuto, the evaluators for those are already in the scripts folder - https://github.com/codelion/optillm/tree/main/scripts

**Comment by itsmeknt** [comment]
I should have time to run the three benchmarks on my rig later this week! I will be using Qwen3 Coder 30B A3B Instruct AWQ 4-bit model, hope that is still useful to you guys.

I was diving into the deepconf implementation code. It seems like you compute the confidence by passing the `logits` into the confidence calculator, and then the confidence calculator converts them into `logprobs` and then compute the confidence scores.

Are there any plans to have deepconf supported for `proxy-only` mode? deepconf seems like a perfect match for SGLang's radix attention mechanism and I'd feel like it could speed up the time a lot using SGLang as the backend. deepconf will take a very long time and I feel like a batching approach + radix attention would be such a nice QoL. I believe all the OpenAI-compatible engines exposes `logprobs` directly so integration should be simple.

**Comment by codelion** [comment]
I think with a 4-bit quantized model it may not work as well since we need to look at the log probs at the very least it will require a lot of tuning of config etc.
